pandas/core/arrays/datetimes.py

```diff
@@ -1652,6 +1654,13 @@ def maybe_convert_dtype(data, copy):
         raise TypeError("Passing PeriodDtype data is invalid.  "
                         "Use `data.to_timestamp()` instead")
 
+    elif is_categorical_dtype(data):
+        # GH#18664 preserve tz in going DTI->Categorical->DTI
+        # TODO: cases where we need to do another pass through this func,
```

Comment:
why is the TODO case here, what is an example? do you have an xfailed test?

Reply:
example is Categorical(TimedeltaIndex) where we should do another pass through maybe_convert_dtype in order to issue the FutureWarning.  I'm assuming there are other corner cases here that will need to be caught/tested.